plugins/module_utils/network/sonic/config/dhcp_snooping/dhcp_snooping.py

```diff
@@ -0,0 +1,646 @@
+        return [{'path': self.binding_uri + '={mac},{ipv}'.format(mac=entry.get('mac_addr'), ipv=afi.get('afi')), 'method': self.delete_method_value}]
+
+    def get_delete_replaced_groupings(self, afis):
+        '''builds list of requests to handle replaced state for both address families'''
+        modified_afi_commands = []
+        requests = []
+
+        want_ipv4 = afis.get('want_ipv4')
+        have_ipv4 = afis.get('have_ipv4')
+        want_ipv6 = afis.get('want_ipv6')
+        have_ipv6 = afis.get('have_ipv6')
+
+        if want_ipv4 and have_ipv4:
+            ipv4_commands, ipv4_requests = self.get_delete_replaced_groupings_afi(want_ipv4, have_ipv4)
+            requests.extend(ipv4_requests)
+            if ipv4_commands:
+                ipv4_commands["afi"] = want_ipv4["afi"]
+                modified_afi_commands.append(ipv4_commands)
+        if want_ipv6 and have_ipv6:
+            ipv6_commands, ipv6_requests = self.get_delete_replaced_groupings_afi(want_ipv6, have_ipv6)
+            requests.extend(ipv6_requests)
+            if ipv6_commands:
+                ipv6_commands["afi"] = want_ipv6["afi"]
+                modified_afi_commands.append(ipv6_commands)
+
+        sent_commands = []
+        if modified_afi_commands:
+            sent_commands = {"afis": modified_afi_commands}
+
+        return sent_commands, requests
+
+    def get_delete_replaced_groupings_afi(self, want_afi, have_afi):
+        '''creates and builds a list of requests to handle all parts that need to be deleted
+        while handling the replaced state for an address family'''
+        sent_commands = {}
+        requests = []
+        diff_requested = get_diff(have_afi, want_afi, self.test_keys)
+
+        if diff_requested.get("vlans") and "vlans" in want_afi:
+            # delete any vlans that are different
+            to_delete = {"afi": have_afi["afi"], "vlans": diff_requested["vlans"]}
+            sent_commands["vlans"] = deepcopy(diff_requested["vlans"])
+            requests.extend(self.get_delete_vlans_requests(to_delete))
+        if diff_requested.get('trusted') and 'trusted' in want_afi:
+            # delete anything that has a difference, covers things that are
+            # in have but not want and things in both but modified
+            to_delete = {"afi": have_afi["afi"], "trusted": diff_requested["trusted"]}
+            sent_commands["trusted"] = deepcopy(diff_requested["trusted"])
+            requests.extend(self.get_delete_trusted_requests(to_delete))
+        if diff_requested.get('source_bindings') and 'source_bindings' in want_afi:
+            # assuming source bindings considered a replaceable subsection ie the list afterwards
+            # should look exactly like what was passed into want
+            if want_afi["source_bindings"] == []:
+                sent_commands["source_bindings"] = deepcopy(have_afi["source_bindings"])
+                requests.extend(self.get_delete_all_source_bindings_request())
```

Comment:
Please update the delete request, to delete only the bindings available in the specified address family.
```suggestion
                requests.extend(self.get_delete_specific_source_bindings_requests(have_afi))
```